tests/integration/suite/actors/reminders/rebalancing.go

```diff
@@ -419,14 +406,19 @@ func (i *rebalancing) reportStatusToPlacement(ctx context.Context, stream placem
 
 	errCh := make(chan error)
 	go func() {
+		var sent bool
 		for {
+			// When the stream ends (which happens when the context is canceled) this returns an error and we can return
 			o, rerr := stream.Recv()
 			if rerr != nil {
-				errCh <- fmt.Errorf("error from placement: %w", rerr)
+				if !sent {
+					errCh <- fmt.Errorf("error from placement: %w", rerr)
+				}
+				return
 			}
-			if o.GetOperation() == "update" {
+			if o.GetOperation() == "update" && !sent {
 				errCh <- nil
-				return
+				sent = true
 			}
 		}
```

Comment:
I do not get a race condition for this test on master.

Reply:
The race condition is due to concurrent `recv` and `send` in the gRPC channel. Not sure why it doesn't appear on master (maybe because the background goroutine was terminated on the first message?) but that's not new code. Fixing that requires creating 2 separate streams and it's a moderately-sized code change. Can be fixed later